src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -105,8 +115,46 @@ protected internal override PinMode GetPinMode(int pinNumber)
 
         protected internal override bool IsPinModeSupported(int pinNumber, PinMode mode)
         {
-            // Libgpiod Api do not support pull up or pull down resistors for now.
-            return mode != PinMode.InputPullDown && mode != PinMode.InputPullUp;
+            switch (mode)
+            {
+                case PinMode.Input:
+                    return true;
+                case PinMode.InputPullDown:
+                case PinMode.InputPullUp:
+                    // for use the bias flags we need libgpiod version 1.5 or later
+                    IntPtr libgpiodVersionPtr = Interop.libgpiod.gpiod_version_string();
+                    string libgpiodVersionMatch = Marshal.PtrToStringAnsi(libgpiodVersionPtr);
+                    Version libgpiodVersion = new Version(libgpiodVersionMatch);
+                    bool isLibgpiod1dot5 = (libgpiodVersion.Major >= 1 && libgpiodVersion.Minor >= 5);
+
+                    // for use the bias flags we need Kernel version 5.5 or later
+                    string[] linuxVersionMatch = RuntimeInformation.OSDescription.Replace("Linux ", string.Empty).Split('.');
```

Comment:
Note that you should also check this string actually contains `Linux` and not something completely different to avoid any weird behaviors in case something changes in the future.

Reply:
Hi @krwq , thanks for the review and tips. I am sorry for the delay in working on it, the quarantine for the COVID-19 started to take effect this week here in the state where I live in Brazil. It was a chaotic week. I hope to be able to work on it this weekend. I hope you are safe. Thanks.

Reply:
Thank you @microhobby for your contribution! 

Question: do we really need to check both libgpiod version and Linux kernel? Could someone install libgpiod  1.5 on linux with kernel lower than 5.5? Isn't enough to check only libgpiod version installed?

Reply:
I have tried install Libgpoid 1.5 on Ubuntu with kernel version 4.4 and got this error: 
`configure: error: "libgpiod needs linux headers version >= v5.5.0"`. Which means we don't need to check kernel version in addition to Libgpoid version, please remove kernel version check

Reply:
Hi @buyaa-n, thanks for the review and checks.

The problem I saw, when I first wrote the patches, was in the case of a cross-compile, I can just use headers from Linux repo and cross compile 1.5 on my machine with no problems. For example in my tests I can use libgpiod 1.5 with an old Kernel version which returns the following error: 

```
Unhandled exception. System.IO.IOException: Error setting pin mode, for pin: 23, error: 22
   at System.Device.Gpio.Drivers.LibGpiodDriver.SetPinMode(Int32 pinNumber, PinMode mode)
``` 
The kernel does not understand the argument that libpgiod sends and then returns an error 22 (Invalid argument). 

The initial idea was to make it more verbose in error handling that kernel 5.5 or later should be used together with libgpiod 1.5. But with your reviews I start to think that it can be adding complexity without much value, this is a tool for developers so, I think that just checking for libgpiod 1.5 or later is enough.

Reply:
I sent two new commits removing the kernel check and improving the libgpiod version check with @krwq suggestions. I wish you are safe and healthy, thanks.

Reply:
Great, thank you for doing this @microhobby, hope you are safe too